Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertSpace.cs

```diff
@@ -0,0 +1,117 @@
+﻿using Autodesk.Revit.DB;
+using Objects.BuiltElements;
+using Objects.Geometry;
+using Speckle.Core.Models;
+using System.Collections.Generic;
+using System.Linq;
+using System;
+using DB = Autodesk.Revit.DB.Mechanical;
+using Point = Objects.Geometry.Point;
+
+namespace Objects.Converter.Revit
+{
+    public partial class ConverterRevit
+    {
+        public List<ApplicationPlaceholderObject> SpaceToNative(Space speckleSpace)
+        {
+            var revitSpace = GetExistingElementByApplicationId(speckleSpace.applicationId) as DB.Space;
+            var level = LevelToNative(speckleSpace.level);
+            var basePoint = PointToNative(speckleSpace.basePoint);
+            var upperLimit = LevelToNative(speckleSpace.upperLimit);
+
+            // create new space if none existing, include zone information if available
+            if (revitSpace == null)
+            {                
+                revitSpace = Doc.Create.NewSpace(level, new UV(basePoint.X, basePoint.Y));
+                if (speckleSpace.zoneId != null)
+                {
+                    var speckleZone = Doc.GetElement(speckleSpace.zoneId) as DB.Zone;
+                    if (speckleZone != null) // else space is added to default zone
+                    {
+                        var spaceSet = new DB.SpaceSet();
+                        spaceSet.Insert(revitSpace);
+                        speckleZone.AddSpaces(spaceSet);
+                    }
+                }
+            }
+            else
+            {                
+                // unplaced space, so just delete and recreate from scratch (but make sure to copy host zone info)
+                // ideally we just place/move the existing space instead, but this seems like a real pain to do with the revit api? should update in future!
+                if (revitSpace.Area == 0 && revitSpace.Location == null) 
+                {
+                    var zone = revitSpace.Zone;
+                    Doc.Delete(revitSpace.Id);
+                    revitSpace = Doc.Create.NewSpace(level, new UV(basePoint.X, basePoint.Y));
+                    if (zone != null)
+                    {
+                        var spaceSet = new DB.SpaceSet();
+                        spaceSet.Insert(revitSpace);
+                        zone.AddSpaces(spaceSet);
+                    }
+                }
+            }
+
+            revitSpace.Name = speckleSpace.name;
+            revitSpace.Number = speckleSpace.number;
+
+            // if user does not specify an UpperLimit level, assume use of default UpperLimit (current level) and default offset values (base offset of 0, limit offset is distance to next level above)
+            if(upperLimit != null)
+            {
+                revitSpace.get_Parameter(BuiltInParameter.ROOM_UPPER_LEVEL).Set(upperLimit.Id); // not sure why, don't actually seem to be able to set the UpperLimit property when UpperLimit is the same as Level - it stays null after assignment!
```

Comment:
Not sure why it doesn't work, I checked and it's the correct param name!
Btw, you can use this to set it directly:

`TrySetParam(revitSpace, BuiltInParameter.ROOM_UPPER_LEVEL, upperLimit);`

Reply:
Thanks, will use that going forward! Still not sure that param is behaving as expected unfortunately? Doesn't seem to be possible to set the value when UpperLimit is null (ie. as with a newly created Space) and the UpperLimit level is to be the same as the Level level: 
![image](https://user-images.githubusercontent.com/69314485/132035076-cba5e634-925c-4a8e-a2aa-c43a3ec81307.png)

What do you think - shall we post on Revit forums?

Reply:
Well, Revit can be v picky when setting parameters. Eg. for columns it wouldn't take offsets unless they are set in a specific order. I guess something similar is at play here!
I'm happy to merge this PR despite this, and I'll leave it up to you to decide whether to post on forums or try to find what makes it work (if anything!)